network/framework/src/application/tests.rs

```diff
@@ -427,6 +427,75 @@ fn test_peers_and_metadata_caching() {
     );
 }
 
+#[tokio::test]
+async fn test_peers_and_metadata_subscriptions() {
```

Comment:
Can we add another test to ensure that notifications are sent for the initial set (if `subscribe()` is called after peers already exist?)

Reply:
added a second `.subscribe()` after one connection to check that we get a NewPeer for the already connected peer.